packages/shared/src/components/post/block/PostTagsPanel.tsx

```diff
@@ -0,0 +1,122 @@
+import CloseButton from '../../CloseButton';
+import { Button, ButtonSize } from '../../buttons/Button';
+import { SourceAvatar } from '../../profile/source';
+import useFeedSettings from '../../../hooks/useFeedSettings';
+import { BlockTagSelection, getBlockedMessage } from './common';
+
+interface PostTagsPanelProps {
+  post: Post;
+  className?: string;
+  toastOnSuccess?: boolean;
+}
+
+export function PostTagsPanel({
+  post,
+  className,
+  toastOnSuccess = true,
+}: PostTagsPanelProps): ReactElement {
+  const { feedSettings } = useFeedSettings();
+  const hasBlockedSource = () =>
+    feedSettings?.excludeSources?.some(({ id }) => id === post.source.id);
+  const [initialPreference] = useState(hasBlockedSource);
+  const [shouldBlockSource, setShouldBlockSource] = useState(hasBlockedSource);
+  const [tags, setTags] = useState<BlockTagSelection>(
+    () =>
+      feedSettings?.blockedTags?.reduce(
+        (block, tag) => ({ ...block, [tag]: post?.tags.includes(tag) }),
+        {},
+      ) ?? {},
+  );
+  const {
+    data: { showTagsPanel, blocked },
+    blockedTags,
+    onClose,
+    onBlock,
+    onReport,
+    onUndo,
+    onDismissPermanently,
+  } = useBlockPost(post, {
+    toastOnSuccess,
+    blockedSource: initialPreference,
+  });
+
+  if (post.tags.length === 0 || isNullOrUndefined(showTagsPanel)) return null;
+
+  if (!showTagsPanel) {
+    if (toastOnSuccess) return null;
+
+    const sourcePreferenceChanged =
+      initialPreference !== blocked?.sourceIncluded;
+    const noAction = blockedTags === 0 && !sourcePreferenceChanged;
+
+    return (
+      <PostBlockedPanel
+        className="mt-6"
+        onActionClick={noAction ? onDismissPermanently : onUndo}
+        message={getBlockedMessage(blockedTags, sourcePreferenceChanged)}
+        ctaCopy={noAction ? `Don't ask again` : 'Undo'}
+      />
+    );
+  }
+
+  return (
+    <div
+      className={classNames(
+        'flex relative flex-col border border-theme-divider-tertiary rounded-16 p-4 pb-0',
+        className,
+      )}
+    >
+      <CloseButton
+        className="top-3 right-3"
+        position="absolute"
+        onClick={() => onClose()}
+        buttonSize={ButtonSize.Small}
+      />
+      <h4 className="font-bold typo-body">Don&apos;t show me posts from...</h4>
+      <p className="mt-1 typo-callout text-theme-label-tertiary">
+        Pick all the topics you are not interested to see on your feed
+      </p>
+      <span
+        className="flex overflow-auto flex-row flex-wrap flex-1 gap-2 content-start mt-4"
+        role="list"
+      >
+        <Button
+          className={shouldBlockSource ? 'btn-primary' : 'btn-tertiaryFloat'}
+          buttonSize={ButtonSize.Small}
+          icon={<SourceAvatar source={post.source} />}
+          onClick={() => setShouldBlockSource(!shouldBlockSource)}
+        >
+          {post.source.name}
+        </Button>
+        {post.tags.map((tag) => (
```

Comment:
Not urgent, but wanted to say we use this often, potentially could be a `TagButton` component.

Reply:
Ah, right. I've been looking for this. I ended up seeing the `TagLinks` which I saw was probably an outdated implementation.

We can definitely go with `TagButton`.